internal/server/receiver_handlers.go

```diff
@@ -378,15 +404,25 @@ func (s *ReceiverServer) annotate(ctx context.Context, resource apiv1.CrossNames
 		return fmt.Errorf("unable to read %s '%s' error: %w", resource.Kind, objectKey, err)
 	}
 
```

Comment:
Need to call `s.annotate` for the single object at the end.

It would be great to add some tests for the function using the fake client.

Reply:
ouch, good 🎣